### DISCUSSION TITLE
upload.render多次触发上传不执行问题

### DISCUSSION THREAD

**Discussion by aaa2281002** [layui/layui#2157] [问答]
upload.render({
            elem: '#iconUpload'
            , multiple: true
            , accept: 'images'
            , exts: 'png|jpg|jpeg'
            , acceptMime: 'image/jpeg,image/png,image/jpg'
            , url: "" //
            , size: 1024*5
            // , number: 2
            , auto: false
            , choose: function (obj) { //选择文件后的回调
                debugger;
//这里也是触发过后再次点击上传也不会触发了
                if(uploadNum > uploadSize){
                    layer.msg("最多上传" + uploadSize + "张图片", {
                        icon: 2,
                        time: 3000
                    });
                    return ;
                }
             
            },
            before: function(obj){
                debugger;

          //触发这个过后 返回false 再次点击上传就不会触发了
                if(1==1){
                    layer.msg("请选择数据来源", {
                        icon: 2,
                        time: 3000
                    });
                    $("input[name='file']").val("");
                    return false;
                }
                this.data = {
                    sources: sources
                }
      
                    return true;
            
            },
            // choose:function (obj){
            //     console.log('2');
            // },
            progress:function (obj){
                debugger;
                // console.log('3');
            },
            allDone:function (obj){
                debugger;
                // console.log('5');
            },
            error:function (index, upload, res){
                debugger;
//这里也是触发过后再次点击上传也不会触发了
                console.log(res);
            }
        });

多处触发一次后，在点击上传就出现无法再次触发上传

**Comment by LSL1618** [comment]
`$("input[name='file']").val("");`这都把选择的文件置空了，可能是这个原因。

**Comment by aaa2281002** [comment]
跟设置空没关系。 重复上传相同路径的文件对象是不行的。


该邮件从移动设备发送


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "layui/layui"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年8月13日(星期二) 上午9:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [layui/layui] upload.render多次触发上传不执行问题 (Discussion #2157)





 
$("input[name='file']").val("");这都把选择的文件置空了，可能是这个原因。

**Reply by LSL1618** [reply]
重复选择同一个文件确实不行，需要在选择之前清除`file input`的`value`值。

**Comment by aaa2281002** [comment]
我清除了啊。 但是还是不行


该邮件从移动设备发送


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "layui/layui"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年8月13日(星期二) 中午12:03
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [layui/layui] upload.render多次触发上传不执行问题 (Discussion #2157)





 
重复选择同一个文件确实不行，需要在选择之前清除file input的value值。

**Reply by LSL1618** [reply]
应该在文件上传完后去清除，before回调方法是上传前，你在上传前就清除了，真正上传的时候文件都没有了还怎么上传！
